safe-modules/liveness-module-v2.md

```diff
@@ -0,0 +1,139 @@
+the existing `LivenessGuard`/`LivenessModule`. `LivenessModuleV2` would employ a challenge/response
+mechanism as an alternative to the existing poke mechanism inside of `LivenessGuard`.
+
+### Singleton
+
+`LivenessModuleV2` is a singleton that exists at a known address on all chains and has no
+constructor parameters. Any Safe on any network can choose to configure the module and then enable
+the module within the Safe.
+
+### Configuration
+
+A Safe configures the module by setting a challenge period, a challenge bond, and a fallback owner.
+
+### Challenge Protocol
+
+Any user may trigger an Account Liveness Challenge by targeting a specific Safe address. In
+response to this challenge, a threshold of owners (same threshold as the Safe itself) must sign a
+message that confirms the liveness of the safe (e.g., a message over the Safe address and the
+current timestamp). If the Safe fails to reach this threshold, all owners are removed from the
+account and ownership is transferred to the fallback account. If the Safe successfully reaches the
+threshold, the challenge fails and the Safe receives the bond as a reward.
+
+The Account Liveness Challenge addresses both of the ways in which a Safe can fail to be live. If
+sufficiently many honest users cannot sign that the account cannot reach a threshold, the account
+falls back to the fallback owner. If sufficiently many users are malicious that the account cannot
+reach a threshold, the honest users can refuse to sign and force a fallback.
+
+## Alternatives Considered
+
+<!-- Describe any alternatives that were considered during the development of this design. Explain
+why the alternative designs were ultimately not chosen and where they failed to meet the product
+requirements. -->
+
+### Owner Liveness Challenge
+
+We previously considered a model that would allow users to challenge either the entire Safe (as
+happens in the Account Liveness Challenge) or an individual owner (i.e., an Owner Liveness
+Challenge). An Owner Liveness Challenge would be more similar to the behavior of the existing
+`LivenessModule`, where a user could target a specific owner and require that owner respond or be
+removed from the Safe.
+
+The Owner Liveness Challenge faces a number of problems:
+
+- It creates a second challenge path.
+- It only solves the honest user case, whereas the Account Liveness Challenge solves both cases.
+- It requires that the Safe configure various new parameters like minimum owner count and threshold
+  percentage which creates more surface area for potential issues.
+
+Additionally, in most cases, the Owner Liveness Challenge would be unused because the Safe would
+prefer to simply remove an owner that is unable to sign.
+
+## Risks and Uncertainties
+
+<!-- Explain any risks and uncertainties that this design includes. Highlight aspects of the design
+that remain unclear and any potential issues we may face down the line. -->
+
```

Comment:
There is the risk that a multisig is challenged and no one notices. We will need monitoring to mitigate that.

Reply:
It is critical that it is monitored, yes.